src/lib/OS/Vma.cc

```diff
@@ -1,99 +1,150 @@
 #include "simeng/OS/Vma.hh"
 
+#include <sys/stat.h>
+
+#include <cstddef>
+
 namespace simeng {
 namespace OS {
+HostFileMMap HostBackedFileMMaps::mapfd(int fd, size_t len, off_t offset) {
+  if (offset & (PAGE_SIZE - 1)) {
+    std::cerr << "[SimEng:HostBackedFileMMaps] Failed to create Host backed "
+                 "file mapping. Offset is not aligned "
+                 "to page size: "
+              << offset << std::endl;
+    std::exit(1);
+  }
+  struct stat* statbuf = (struct stat*)malloc(sizeof(struct stat));
+  int fstatResult = fstat(fd, statbuf);
+  if (fstatResult < 0) {
+    std::cerr << "[SimEng:HostBackedFileMMaps] fstat failed: Cannot create "
+                 "host backed file mmap for file "
+                 "descriptor - "
+              << fd << std::endl;
+    free(statbuf);
+    std::exit(1);
+  }
+  off_t fstatFileSize = statbuf->st_size;
+  free(statbuf);
```

Comment:
Here you free(statbuf) but you then use it again during the `mmap` call and when passed to `hostFileMMap`. Shouldn't this free be moved to the end of this function?

Reply:
corrected